src/webgl/primitives.js

```diff
@@ -800,49 +801,115 @@ p5.RendererGL.prototype.triangle = function(args) {
 };
 
 p5.RendererGL.prototype.ellipse = function(args) {
-  var x = args[0];
-  var y = args[1];
-  var width = args[2];
-  var height = args[3];
-  //detailX and Y are optional 6th & 7th
-  //arguments
-  var detailX = args[4] || 24;
-  var gId = 'ellipse|' + detailX;
+  this.arc(
+    args[0],
+    args[1],
+    args[2],
+    args[3],
+    0,
+    constants.PI * 2,
+    constants.OPEN,
+    args[4]
+  );
+};
+
+p5.RendererGL.prototype.arc = function(args) {
+  var x = arguments[0];
+  var y = arguments[1];
+  var width = arguments[2];
+  var height = arguments[3];
+  var start = arguments[4];
+  var stop = arguments[5];
+  var mode = arguments[6];
+  var detail = arguments[7] || 25;
+
+  var shape;
+  var gId;
+
+  if (Math.abs(stop - start) >= constants.PI * 2) {
+    shape = 'ellipse';
+    gId = shape + '|' + x + '|' + y + '|' + width + '|' + height + '|';
```

Comment:
you can remove x,y, width and height here because they are part of the model scaling on line 915.